projects/wallet-template/src/background/keystore/types.ts

```diff
@@ -1,10 +1,28 @@
-import { Keyset } from "../types"
 import { KeystoreV4 } from "./keystoreV4"
 
-type KeystoreMeta = {
-  // TODO: support single account
-  type: "keyset"
-} & Keyset
+export type DerivationPath = {
+  chainId: string
+  path: string
+  publicKey: string
+}
+
+export type BaseKeystore = {
+  name: string
+  scheme: "Sr25519" | "Ed25519" | "Ecdsa"
+  createdAt: number
+}
+
+export type DerivationPathKeystore = BaseKeystore & {
+  _type: "DerivationPathKeystore"
+  derivationPaths: DerivationPath[]
+}
+
+export type PrivateKeyKeystore = BaseKeystore & {
+  _type: "PrivateKeyKeystore"
+  privateKey: string
```

Comment:
see previous comment, the `privateKey` should be encrypted and removed from the keystore meta.

And the `publicKey` should be precomputed and stored in the keystore meta